### DISCUSSION TITLE
Script query_params question

### DISCUSSION THREAD

**Discussion by rowco** [netbox-community/netbox#13673] [General]
Hi,

I've written a custom script which requires a device and an interface for the input. The devices are all members of virtual chassis, but I only want to show the interfaces from the specified device. When I use the following I get the interfaces from both devices in the virtual chassis. 

```   
    device = ObjectVar(
        model=Device,
        query_params={
            'status': 'active'
        }
    )

    interfaces = ObjectVar(
        model=Interface,
        query_params={
            'device_id': '$device'
        }
    )
```

**Comment by kkthxbye-code** [comment]
From the 3.6.0 release notes:

> The device and device_id filter for interfaces will no longer include interfaces from virtual chassis peers. Two new filters, virtual_chassis_member and virtual_chassis_member_id, have been introduced to match all interfaces belonging to the specified device's virtual chassis (if any).

> https://github.com/netbox-community/netbox/issues/11478 - Introduce virtual_chassis_member filter for interfaces & restore default behavior for device filter

https://github.com/netbox-community/netbox/releases/tag/v3.6.0

It was a bug introduced a while ago, it got fixed i 3.6.0. Not sure if there's a workaround.

**Reply by rowco** [reply]
Ah perfect, upgrade time :)
